ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -178,7 +178,7 @@ void addJoin(final JoinInfo joinInfo) {
   }
 
   public List<JoinInfo> getJoin() {
-    return joinInfo;
+    return Collections.unmodifiableList(joinInfo);
```

Comment:
Cannot use `ImmutableList` as `joinInfo` is build up incrementally (same for `tableFunctions` below)`